packages/purpose-process/src/services/purposeService.ts

```diff
@@ -733,6 +738,114 @@ export function purposeServiceBuilder(
+
+      if (purposeIsDraft(purposeToClone.data)) {
+        throw purposeCannotBeCloned(purposeId);
+      }
+
+      const dailyCalls = getDailyCallsFromPurposeToClone(purposeToClone.data);
+
+      const newPurposeVersion: PurposeVersion = {
+        id: generateId(),
+        createdAt: new Date(),
+        state: purposeVersionState.draft,
+        dailyCalls,
+      };
+
+      const riskAnalysisFormToClone = purposeToClone.data.riskAnalysisForm;
+
+      const clonedRiskAnalysisForm: PurposeRiskAnalysisForm | undefined =
+        riskAnalysisFormToClone
+          ? {
+              id: generateId(),
+              version: riskAnalysisFormToClone.version, // TO DO double-check
+              riskAnalysisId: riskAnalysisFormToClone.riskAnalysisId,
+              singleAnswers: riskAnalysisFormToClone.singleAnswers.map(
+                (answer) => ({
+                  ...answer,
+                  id: generateId(),
+                })
+              ),
+              multiAnswers: riskAnalysisFormToClone.multiAnswers.map(
+                (answer) => ({
+                  ...answer,
+                  id: generateId(),
+                })
+              ),
+            }
+          : undefined;
+
+      const currentDate = new Date();
+      const title = purposeToClone.data.title;
+      const suffix = ` - clone - ${formatDateAndTime(currentDate)}`;
+      const prefixLengthAllowance = 60 - suffix.length - 3;
+      // 60 is the maximum length for the purpose title, according to the api spec (PurposeSeed)
+      const clonedPurposeTitle =
+        title.length + suffix.length <= 60
+          ? `${title}${suffix}`
+          : `${title.slice(0, prefixLengthAllowance)}...${suffix}`;
+
+      await assertPurposeTitleIsNotDuplicated({
+        readModelService,
+        eserviceId: unsafeBrandId(seed.eserviceId),
+        consumerId: organizationId,
+        title: clonedPurposeTitle,
+      });
+
+      const clonedPurpose: Purpose = {
+        title: clonedPurposeTitle,
+        id: generateId(),
+        createdAt: currentDate,
+        eserviceId: unsafeBrandId(seed.eserviceId),
+        consumerId: organizationId,
+        description: purposeToClone.data.description,
+        versions: [newPurposeVersion],
+        isFreeOfCharge: purposeToClone.data.isFreeOfCharge,
+        freeOfChargeReason: purposeToClone.data.freeOfChargeReason,
+        riskAnalysisForm: clonedRiskAnalysisForm,
+      };
+
+      const isRiskAnalysisValid = clonedRiskAnalysisForm
+        ? validateRiskAnalysis(
+            riskAnalysisFormToRiskAnalysisFormToValidate(
+              clonedRiskAnalysisForm
+            ),
+            false,
+            tenant.kind
+          ).type === "valid"
+        : false;
+
+      const event = toCreateEventPurposeCloned({
+        purpose: clonedPurpose,
+        sourcePurposeId: purposeToClone.data.id,
+        sourceVersionId: generateId(), // TO DO not sure where to take this, but the event definition requires it
```

Comment:
this is the version the daily calls are taken from

Reply:
Makes sense, updated.